File: update_colors.py
import os
import re

src_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'src')

replacements = [
    # Burgundy / Chocolate -> Green-Patina Enamel (#3C5B42) & variations
    (re.compile(r'3a1e26', re.IGNORECASE), '3C5B42'),
    (re.compile(r'2a151b', re.IGNORECASE), '2E4633'),
    (re.compile(r'522a36', re.IGNORECASE), '4A6D51'),
    (re.compile(r'5a2e38', re.IGNORECASE), '4A6D51'),
    (re.compile(r'4a252f', re.IGNORECASE), '3C5B42'),
    (re.compile(r'8b1b36', re.IGNORECASE), '3C5B42'),
    (re.compile(r'991b36', re.IGNORECASE), '3C5B42'),
    (re.compile(r'7a172b', re.IGNORECASE), '2E4633'),
    (re.compile(r'631323', re.IGNORECASE), '2E4633'),

    # Vibrant Pink / Crimson -> Red Ribbon Fabric (#B63737) & variations
    (re.compile(r'92141f', re.IGNORECASE), 'B63737'),
    (re.compile(r'92141f', re.IGNORECASE), '962D2D'),
    (re.compile(r'e11e5c', re.IGNORECASE), 'C74A4A'),
    (re.compile(r'ce1650', re.IGNORECASE), 'B63737'),
    (re.compile(r'c21153', re.IGNORECASE), '962D2D'),
    (re.compile(r'a00d43', re.IGNORECASE), '802525'),
    (re.compile(r'8b0a3c', re.IGNORECASE), '802525'),

    # Luxury Gold / Amber -> Aged Brass Metalwork (#A4936B) & Lightest Highlight (#BFB394)
    (re.compile(r'ffd700', re.IGNORECASE), 'A4936B'),
    (re.compile(r'fbbf24', re.IGNORECASE), 'BFB394'),
    (re.compile(r'f59e0b', re.IGNORECASE), 'A4936B'),
    (re.compile(r'd97706', re.IGNORECASE), 'A4936B'),
    (re.compile(r'b45309', re.IGNORECASE), '827453'),
    (re.compile(r'f57c00', re.IGNORECASE), 'A4936B'),
    (re.compile(r'eab308', re.IGNORECASE), 'BFB394'),

    # Light Backgrounds -> Neutral White (#FFFFFF) & Cream White (#E8E2CF)
    (re.compile(r'fdfaf5', re.IGNORECASE), 'E8E2CF'),
    (re.compile(r'FAF5E8', re.IGNORECASE), 'E8E2CF'),
    (re.compile(r'fef9f3', re.IGNORECASE), 'E8E2CF'),
    (re.compile(r'fdfaf6', re.IGNORECASE), 'E8E2CF'),
    (re.compile(r'F5F5F5', re.IGNORECASE), 'FFFFFF'),
    (re.compile(r'F9FAFB', re.IGNORECASE), 'FFFFFF'),
    (re.compile(r'F3F4F6', re.IGNORECASE), 'E8E2CF')
]

updated_count = 0
for root, dirs, files in os.walk(src_dir):
    for file in files:
        if file.endswith(('.jsx', '.js', '.css', '.html')):
            file_path = os.path.join(root, file)
            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            modified = content
            for regex, repl in replacements:
                modified = regex.sub(repl, modified)
                
            if content != modified:
                with open(file_path, 'w', encoding='utf-8') as f:
                    f.write(modified)
                print(f"Updated: {os.path.relpath(file_path, src_dir)}")
                updated_count += 1

print(f"Color update complete! Total files updated: {updated_count}")

File: update_colors.js
const fs = require('fs');
const path = require('path');

const srcDir = path.join(__dirname, 'src');

const replacements = [
  // 1. Vibrant Pinks / Magenta / Crimson / Reds -> Racing Red (#92141f) & Dark Racing Red (#b30c15)
  [/92141f/gi, '92141f'],
  [/92141f/gi, 'b30c15'],
  [/e11e5c/gi, '92141f'],
  [/ce1650/gi, 'c70e18'],
  [/c21153/gi, 'b30c15'],
  [/a00d43/gi, '8a080f'],
  [/8b0a3c/gi, '8a080f'],
  [/9e2a4b/gi, '92141f'],
  [/800020/gi, 'b30c15'],
  [/6d1f3b/gi, 'b30c15'],
  [/86284a/gi, '92141f'],
  [/73223f/gi, 'b30c15'],
  [/5a1930/gi, '8a080f'],
  [/from-pink-400/gi, 'from-[#92141f]'],
  [/to-pink-500/gi, 'to-[#b30c15]'],
  [/pink-400/gi, 'red-500'],
  [/pink-500/gi, 'red-700'],

  // 2. Dark Burgundy / Chocolate / Deep Brown -> British Racing Green Dark (#044d1d) & Deep Forest (#033815)
  [/3a1e26/gi, '044d1d'],
  [/2a151b/gi, '033815'],
  [/522a36/gi, '065f24'],
  [/5a2e38/gi, '065f24'],
  [/4a252f/gi, '044d1d'],
  [/8b1b36/gi, '044d1d'],
  [/991b36/gi, '044d1d'],
  [/7a172b/gi, '033815'],
  [/631323/gi, '033815'],
  [/3d1a29/gi, '033815'],

  // 3. Gold / Yellow / Amber Accents -> British Racing Green Light (#d0dbd4) & Sleek Highlights
  [/ffd700/gi, 'd0dbd4'],
  [/fbbf24/gi, 'd0dbd4'],
  [/f59e0b/gi, 'd0dbd4'],
  [/d97706/gi, 'd0dbd4'],
  [/eab308/gi, 'd0dbd4'],
  [/f57c00/gi, 'd0dbd4'],
  [/c69214/gi, '044d1d'],
  [/b8860b/gi, '044d1d'],
  [/b45309/gi, '044d1d'],

  // 4. Warm Cream / Off-White Light Backgrounds -> Theme Light (#F2F2F2) & Pure White (#FFFFFF)
  [/fdfaf5/gi, 'F2F2F2'],
  [/FAF5E8/gi, 'F2F2F2'],
  [/fef9f3/gi, 'F2F2F2'],
  [/fdfaf6/gi, 'F2F2F2'],
  [/fff8ed/gi, 'F2F2F2'],
  [/fcf9f2/gi, 'F2F2F2'],
  [/F3F4F6/gi, 'F2F2F2'],
  [/F5F5F5/gi, 'F2F2F2'],
  [/F9FAFB/gi, 'F2F2F2']
];

function processDirectory(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      processDirectory(fullPath);
    } else if (file.endsWith('.jsx') || file.endsWith('.js') || file.endsWith('.css') || file.endsWith('.html')) {
      let content = fs.readFileSync(fullPath, 'utf8');
      let modified = content;
      for (const [regex, replacement] of replacements) {
        modified = modified.replace(regex, replacement);
      }
      if (content !== modified) {
        fs.writeFileSync(fullPath, modified, 'utf8');
        console.log(`Updated colors in: ${path.relative(__dirname, fullPath)}`);
      }
    }
  }
}

processDirectory(srcDir);
console.log('Color update complete!');
